### PR TITLE
Feature/tulcdm 141a ensure assets handled similar to restricted access

### PR DESCRIPTION
**Pull request by skng5**
Ready for review. Merge if it looks good.
- Visibility of private collection is defined with a [literal](https://github.com/tulibraries/tul_cdm/blob/feature/TULCDM-141a-Ensure_assets_handled_similar_to_restricted_access/app/controllers/application_controller.rb#L22). Should we add a list of groups allowed access to the private collection? If so, then should it be part of this issue?
- In RSpec tests, cannot pass authenticated session to application_controller methods. Seems to work in development. Considering moving these methods into an ActiveSupport::Concern module.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -12,12 +14,46 @@ class ApplicationController < ActionController::Base
 
   # Determine if the collection is viewable depending on private flag and white-listed remote IP address
   def is_viewable? (collection)
-    !collection.is_private || (collection.allowed_ip_addresses.split(%r{,\s*}).include? request.remote_ip)
+
+    # If no ip address filtering
+    if (collection['allowed_ip_addresses'].empty?)
+      if (is_private?(collection))
+        # TODO: Brittle - Allow better control over who may view a private collecionts
+        RoleMapper.roles(current_user.email).include? "archivist"
```
Comment (bibliotechy) [inline app/controllers/application_controller.rb]: @skng5 I need a little help understanding this line. Why are we checking if the email of the current user contains the string "archivist"? Is that a default email address of some kind for admin users?

Comment (skng5) [inline app/controllers/application_controller.rb]: Email is passed to RoleMapper.roles method to get the Role mapping in the config/role_map.yml. current_user is the user that's logged into the app by way of http://locahost:3000/users/sign_in.
- Note - I updated the PR description with an additional item to discuss.

**Comment by bibliotechy** [comment]
@skng5 OK, that makes sense. Thanks. And you note above it highlights my actual concern, I think.

**Comment by bibliotechy** [comment]
@skng5 Trying to suss out why functions like `is_viewable?` and `unallowed_by_ip_collections` are in both applications_controller and tul_cdm_helper. It looks like a direct copy paste.

**Comment by skng5** [comment]
@bibliotechy : That was operator error. I started moving the code to the helper with the intention of getting it out of the application controller, but decided to keep it in application controller before I issued the PR. I forgot to revert the helper. I'd like to spend more time with this issue to research moving it out of the application controller into either the helper or a concern.

**Comment by bibliotechy** [comment]
@skng5 That sounds like a good idea. Let's talk tomorrow about the scope of that.

**Comment by skng5** [comment]
In  lib/tul_cdm/models/base.rb I added Hydra::AccessControls which controls show and search results in catatalog_controller with:

``` ruby
  # These before_filters apply the hydra access controls
  before_filter :enforce_show_permissions, :only=>:show
  # This applies appropriate access controls to all solr queries
  CatalogController.solr_search_params_logic += [:add_access_controls_to_solr_params]
```

But this requires us to set permission on ingested objects. This adds complexity to the ingest process. I would like to remove this add-on and write our own custom enforce_show_permissions and a add_access_controls_to_solr that uses the access control we've already created to gate access to Digital Collections.

**Comment by skng5** [comment]
I've made the changes above. I've removed Hydra::AccessControl and it depends solely on access controls in the Digital Collections model. Please review and merge if you think it's ready.

**Comment by skng5** [comment]
To try this feature, use the `seed.rb` file, and set the `is_private` field to `true` of one of the collections.

**Comment by bibliotechy** [comment]
@skng5 Where are IP address ranges for each collection coming from? Are we going to have to define them in a config / seed file? So every time it is updated on ContentDM, it will also have to be updated here? Seems likely to result in config drift between systems.

**Comment by skng5** [comment]
@bibliotechy The IP address ranges will come from the stakeholders. They can be set in the seed.rb, but best practices dictates that the information be entered and modified using the digital collection admin page. The seed should only be run once, when the system is initially started and the database backed up on a regular basis.

**Pull request closed without merging**